python/cudf/cudf/core/reshape.py

```diff
@@ -76,23 +76,25 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
     ignore_index : bool, default False
         Set True to ignore the index of the *objs* and provide a
         default range index instead.
+    sort : bool, default False
+        Sort non-concatenation axis if it is not already aligned.
 
     Returns
     -------
     A new object of like type with rows from each object in ``objs``.
     """
 
-    if sort not in (None, False):
-        raise NotImplementedError("sort parameter is not yet supported")
-
     if not objs:
         raise ValueError("No objects to concatenate")
 
     objs = [obj for obj in objs if obj is not None]
 
     # Return for single object
     if len(objs) == 1:
-        return objs[0]
+        result = objs[0]
+        if ignore_index:
+            result._index = cudf.RangeIndex(len(result))
```

Comment:
This modifies the passed in object which is unexpected behavior. We'd need to at least make a shallow copy and more likely Pandas semantics are a deep copy.